requirements.txt

```diff
@@ -1,58 +1,58 @@
 oic==1.6.1                                                  # for authentication via OpenID Connect protocol
-prometheus_client==0.19.0                                   # Python client for the Prometheus monitoring system
-boto3==1.29.5                                               # S3 boto protocol (new version)
+prometheus_client==0.20.0                                   # Python client for the Prometheus monitoring system
+boto3==1.34.87                                               # S3 boto protocol (new version)
 
 # All dependencies needed in extras for rucio server/daemons should be defined here
 cx_oracle==8.3.0                                            # oracle_extras
 psycopg2-binary==2.9.9                                      # postgresql_extras
 PyMySQL==1.1.0                                              # mysql_extras
 PyYAML==6.0.1                                               # globus_extras and used for reading test configuration files
-globus-sdk==3.32.0                                          # globus_extras
+globus-sdk==3.40.0                                          # globus_extras
 python3-saml==1.16.0                                        # saml_extras
 pymongo==4.6.3                                              # pymongo (metadata plugin)
 libtorrent==2.0.9                                           # Support for the bittorrent transfertool
-qbittorrent-api==2023.11.57                                 # qBittorrent plugin for the bittorrent tranfsertool
+qbittorrent-api==2024.3.60                                 # qBittorrent plugin for the bittorrent tranfsertool
 
 # All dependencies needed to develop/test rucio should be defined here
-pytest==7.4.3
+pytest==8.1.1
 pytest-xdist~=3.5.0                                         # Used for parallel testing
-pyflakes==3.1.0                                             # Passive checker of Python programs
-flake8==6.1.0                                               # Wrapper around PyFlakes&pep8; python_version < '3.9'
-pycodestyle==2.11.0                                         # New package replacing pep8; python_version < '3.9'
-pylint==3.0.2
-astroid==3.0.1
-virtualenv==20.24.7                                         # Virtual Python Environment builder
-xmlsec==1.3.13                                              # Required to install pyproject.toml-based projects
+pyflakes==3.2.0                                             # Passive checker of Python programs
+flake8==7.0.0                                               # Wrapper around PyFlakes&pep8; python_version < '3.9'
+pycodestyle==2.11.1                                         # New package replacing pep8; python_version < '3.9'
+pylint==3.1.0
+astroid==3.1.0
+virtualenv==20.25.3                                         # Virtual Python Environment builder
+xmlsec==1.3.14                                              # Required to install pyproject.toml-based projects
```

Comment:
`xmlsec==1.3.14` should be avoided due to this issue we had: https://github.com/rucio/rucio/issues/6694. Let's not merge this until we figure out a way to tell dependabot to skip this version (I think the tests on this PR will probably fail due to this `xmlsec` version)

Reply:
It will just re-create this PR at 11:30 CEST today (https://github.com/rucio/rucio/blob/master/.github/dependabot.yml#L9-L16) - I'm going to modify the requirements.txt to tell dependabot to skip `xmlsec==1.3.14`. I'm also seeing some major version bumps that might be problematic (e.g. `urllib3` 1.x -> 2.x) - I assume that it should be alright as long as CI/CD passes, but we might still want to pay attention to any possible issues that might occur.